.teamcity/_self/PluginBaseBuild.kt

```diff
@@ -59,6 +59,20 @@ open class PluginBaseBuild : Template({
 		bashNodeScript {
 			name = "Prepare environment"
 			scriptContent = """
+				# Merge the trunk branch first. This way, our builds and tests
+				# include the latest merged version of the plugin being built.
+				# Otherwise, we can get into a situation where the current plugin
+				# build appears "different", but that's just because it's older.
+				if [[ "%teamcity.build.branch.is_default%" != "true" ]] ; then
+					# git operations will fail if no user is set.
+					git config --local user.email "tcbuildagent@example.com"
+					git config --local user.name "Teamcity Build Agent"
+					git checkout -b tc-temp-branch
```

Comment:
Why we need to change to a new branch?

I'm a bit worried about this, it is effectively changes the local name of the branch. How would it affect checks like `%teamcity.build.branch.is_default%` or values in `%teamcity.build.branch%`?

Reply:
My impression was that TeamCity runs in a detached-head state, which I thought meant would couldn't run branch operations like merge -- those might be wrong assumptions, though.

> How would it affect checks like `%teamcity`...

I've at least tested that it at least doesn't have an affect on `%build.vcs.number%` (the commit hash). My impression is that these values are populated statically before the build starts, so we can't modify them from the build steps.

I'll revisit my assumptions here. Not having to use a local test branch would be simpler

Reply:
> My impression was that TeamCity runs in a detached-head state

I was wrong, `git status` reports that it is running on the branch in question. I think I may have mixed this up with a previous CI provider which did run in detached head :) So we don't need a new branch.